include/chainbase/chainbase.hpp

```diff
@@ -933,19 +941,19 @@ namespace chainbase {
          }
 
          template<typename ObjectType, typename Modifier>
-         void modify( const ObjectType& obj, Modifier&& m )
+         void modify( const ObjectType& obj, Modifier&& m, bool move_old_value = false )
```

Comment:
I'm not sure I like the readability of overloading `::modify` to do potentially destructive things before the modifier is invoked based on a boolean flag.

What do you think about leaving `::modify` as it was (makes a copy for the backup always) and adding a  method `::replace` which will move into `head.old_values` always.  

The expected `::replace` method could take a "Replacer" lambda with the signature `void (value_type& value_to_replace, const value_type& old_value)`.  Like `emplace` it would construct the new value and pass it to the lambda which can fully initialize the replacement using the immutable old value if needed.  Boost provides  https://www.boost.org/doc/libs/1_67_0/libs/multi_index/doc/reference/ord_indices.html#replace which has an overload to take a rvalue.  I assume this allows us to `move` the newly constructed replacement into the multi-index container.

Reply:
replace() added, modify() is now the original behavior.